p2p/src/net/libp2p/backend.rs

```diff
@@ -264,12 +252,18 @@ impl Backend {
                     source,
                     result
                 );
-                self.swarm.behaviour_mut().gossipsub.report_message_validation_result(
+                match self.swarm.behaviour_mut().gossipsub.report_message_validation_result(
                     &message_id,
                     &source,
                     result,
-                );
-                response.send(Ok(())).map_err(|_| P2pError::ChannelClosed)
+                ) {
+                    Ok(_) => response.send(Ok(())).map_err(|_| P2pError::ChannelClosed),
+                    Err(e) => response
+                        .send(Err(P2pError::Libp2pError(Libp2pError::PublishError(
+                            e.to_string(),
+                        ))))
+                        .map_err(|_| P2pError::ChannelClosed),
```

Comment:
As I'm seeing "ChannelClosed" all over the place, I'm wondering whether this error can be more expressive by containing the reason of that so that it can go to logs with that information.

Reply:
There isn't a multitude of reasons for why the channel was closed though. It's that the the sender/receiver end was dropped, meaning of the manager shut down for whatever reason.